lib/models/room.js

```diff
@@ -254,39 +223,59 @@ Room.prototype._fixUpLegacyTimelineFields = function() {
     // and this.oldState and this.currentState as references to the
     // state at the start and end of that timeline. These are more
     // for backwards-compatibility than anything else.
-    this.timeline = this._liveTimeline.getEvents();
-    this.oldState = this._liveTimeline.getState(EventTimeline.BACKWARDS);
-    this.currentState = this._liveTimeline.getState(EventTimeline.FORWARDS);
+    this.timeline = this.getUnfilteredTimelineSet().getLiveTimeline().getEvents();
+    this.oldState = this.getUnfilteredTimelineSet().getLiveTimeline()
+                        .getState(EventTimeline.BACKWARDS);
+    this.currentState = this.getUnfilteredTimelineSet().getLiveTimeline()
+                            .getState(EventTimeline.FORWARDS);
 };
 
 /**
- * Get the timeline which contains the given event, if any
+ * Return the timeline sets for this room.
+ * @return {EventTimelineSet[]} array of timeline sets for this room
+ */
+Room.prototype.getTimelineSets = function() {
+    return this._timelineSets;
+};
+
+/**
+ * Helper to return the main unfiltered timeline set for this room
+ * @return {EventTimelineSet} room's unfiltered timeline set
+ */
+Room.prototype.getUnfilteredTimelineSet = function() {
+    return this._timelineSets[0];
+};
+
+/**
+ * Get the timeline which contains the given event from the unfiltered set, if any
  *
  * @param {string} eventId  event ID to look for
  * @return {?module:models/event-timeline~EventTimeline} timeline containing
  * the given event, or null if unknown
  */
 Room.prototype.getTimelineForEvent = function(eventId) {
-    var res = this._eventIdToTimeline[eventId];
-    return (res === undefined) ? null : res;
+    return this.getUnfilteredTimelineSet().getTimelineForEvent(eventId);
+};
+
+/**
+ * Add a new timeline to this room's unfiltered timeline set
+ *
+ * @return {module:models/event-timeline~EventTimeline} newly-created timeline
+ */
+Room.prototype.addTimeline = function() {
```

Comment:
that's just a unit test, isn't it?

I can't see any reason why an application should be adding a timeline to the Room; it seems like the sort of thing that would break things. I would encourage vaping it.